File: ina_2sem/labor_cs/3_lab/2_task/Prostokat.cpp
#include "Prostokat.h"

Prostokat::Prostokat(double a, double b, double kat) 
    : Czworokat(a, b, a, b, kat) {
    if (kat != 90) throw std::invalid_argument("Kąt w prostokącie musi wynosić 90°.");
}

double Prostokat::obliczPole() {
    return bok1 * bok2;
}

double Prostokat::obliczObwod() {
    return 2 * (bok1 + bok2);
}

std::string Prostokat::podajNazwe() {
    return "Prostokąt";
}

File: ina_2sem/labor_cs/3_lab/1_task/Prostokat.java
public class Prostokat extends Czworokat {
    public Prostokat(double a, double b, double kat) {
        super(a, b, a, b, kat);
        if (kat != 90) throw new IllegalArgumentException("Kąt w prostokącie musi wynosić 90°.");
    }

    @Override
    public double obliczPole() {
        return bok1 * bok2;
    }

    @Override
    public double obliczObwod() {
        return 2 * (bok1 + bok2);
    }

    @Override
    public String podajNazwe() {
        return "Prostokąt";
    }
}
